File: src/misc/ActiveDetail.vb
Imports System.ComponentModel
Imports DCS.Quote.Model
Imports DCS.Quote.Common

Public Class ActiveDetail
    Implements INotifyPropertyChanged

    Private _Detail As Detail

    Public Shared ReadOnly ActiveDetail As ActiveDetail = New ActiveDetail

    Public Event PropertyChanged As PropertyChangedEventHandler _
        Implements INotifyPropertyChanged.PropertyChanged

    Public Property Detail As Detail
        Get
            Return _Detail
        End Get
        Friend Set(ByVal value As Detail)
            If (_Detail IsNot value) Then
                _Detail = value
                RaiseEvent PropertyChanged(Me, _
                    New PropertyChangedEventArgs("Detail"))
            End If
        End Set
    End Property

End Class

File: trunk/quote/Projects/PluginModel/ActiveDetail.cs
namespace Model
{
    using System;
    using System.ComponentModel;
    using System.Linq;
    
    using Model.Common;

    public class ActiveDetail : INotifyPropertyChanged
    {
        private static ActiveDetail instance = new ActiveDetail();

        private Detail detail;

        public event PropertyChangedEventHandler PropertyChanged;

        public static object Instance 
        {
            get { return instance; }
        }

        public Detail Detail 
        {
            get 
            { 
                return this.detail; 
            }
            
            set 
            {
                if (!object.ReferenceEquals(this.detail, value))
                {
                    this.detail = value;
                    if (this.PropertyChanged != null) 
                    {
                        this.PropertyChanged(this, new PropertyChangedEventArgs("Detail"));
                    }
                }
            }
        }
    }
}
